subprojects/core/src/main/java/org/gradle/internal/service/scopes/WorkerSharedProjectScopeServices.java

```diff
@@ -157,6 +159,6 @@ ObjectFactory createObjectFactory(InstantiatorFactory instantiatorFactory, Servi
     @Provides
     DefaultProjectLayout createProjectLayout(FileResolver fileResolver, FileCollectionFactory fileCollectionFactory, TaskDependencyFactory taskDependencyFactory,
                                              FilePropertyFactory filePropertyFactory, Factory<PatternSet> patternSetFactory, PropertyHost propertyHost, FileFactory fileFactory) {
-        return new DefaultProjectLayout(projectDir, fileResolver, taskDependencyFactory, patternSetFactory, propertyHost, fileCollectionFactory, filePropertyFactory, fileFactory);
+        return new DefaultProjectLayout(settingsDir, projectDir, fileResolver, taskDependencyFactory, patternSetFactory, propertyHost, fileCollectionFactory, filePropertyFactory, fileFactory);
```

Comment:
❓  We went to great lengths to make the settings directory available as part of the `ProjectLayout` when it's injected in the worker project-scope. Do we have a test that uses that settings directory from the worker?

I might have missed it, but if we don't have it, we should definitely add it

Reply:
Turns out that `ProjectLayout` is only injected to internal code and there for the wrong reasons too... Will revert most of the previous changes in this direction, they are not necessary.